test/e2e/test_job_submissions.py

```diff
@@ -1727,3 +1685,54 @@ def session_action_has_expected_progress(session_action_id) -> bool:
         job.wait_until_complete(client=deadline_client)
 
         assert job.task_run_status == TaskStatus.SUCCEEDED
+
+    def test_worker_enters_stopping_state_while_draining(
+        self,
+        deadline_resources: DeadlineResources,
+        deadline_client: DeadlineClient,
+        function_worker: EC2InstanceWorker,
+        sleep_script: str = (
+            """
+            #!/usr/bin/env bash
+            sleep 600
+            """
+            if os.environ["OPERATING_SYSTEM"] == "linux"
+            else """
+            Start-Sleep -Seconds 600
+            """
+        ),
+    ):
+
+        submit_custom_job(
+            job_name="Infinite Sleep Job",
+            deadline_client=deadline_client,
+            farm=deadline_resources.farm,
+            queue=deadline_resources.queue_a,
+            run_script=sleep_script,
+        )
+
+        if os.environ["OPERATING_SYSTEM"] == "linux":
+            cmd_result = function_worker.send_command("sudo systemctl stop deadline-worker")
+        else:
+            cmd_result = function_worker.send_command("sc.exe stop DeadlineWorker")
+
+        assert cmd_result.exit_code == 0
+
+        @backoff.on_predicate(
+            wait_gen=backoff.constant,
+            max_time=120,
+            interval=10,
+        )
+        def worker_stop(worker: EC2InstanceWorker) -> bool:
```

Comment:
```suggestion
        def is_worker_stopped(worker: EC2InstanceWorker) -> bool:
```

nit: more descriptive function name